app/Http/Controllers/AssetModelsController.php

```diff
@@ -173,8 +179,17 @@ public function update(ImageUploadRequest $request, $modelId = null)
                 }
             }
         }
-
+       
+      
+       
         if ($model->save()) {
+            if ($model->wasChanged('eol')) {
+                    $newEol = $model->eol; 
+                    $model->assets()->whereNotNull('purchase_date')->where('eol_explicit', false)
+                        //this DB::raw is so that we can use an ->update() which is _much_ faster than foreaching
+                        //but, laravel doesn't have a way to access a column inside an update 
+                        ->update(['asset_eol_date' => DB::raw('DATE_ADD(purchase_date, INTERVAL :newEol MONTH)', ['newEol' => $newEol])]); 
```

Comment:
Something is up with this line. I got the following exception when I updated the `EOL` field:
```
SQLSTATE[HY093]: Invalid parameter number: mixed named and positional parameters (SQL: update `assets` set `asset_eol_date` = DATE_ADD(purchase_date, INTERVAL :newEol MONTH), `assets`.`updated_at` = 2023-09-19 18:02:36 where `assets`.`model_id` = 4 and `assets`.`model_id` is not null and `purchase_date` is not null and `eol_explicit` = 0 and `assets`.`deleted_at` is null)
```

PHPStorm also has `['newEol' => $newEol]` grayed out saying `Method call is provided 2 parameters, but the method signature uses 1 parameters`. It looks like `DB::raw()` only takes one parameter.

Reply:
hm, i could've sworn that was working before. i ended up removing the variable thing and just concatenating it, the value is already validated by the time this happens so should be fine.